Doras/src/app/roles/roles.component.ts

```diff
@@ -26,7 +29,17 @@ export class RolesComponent implements OnInit {
     return this.dataTransferService.capability;
 }
 
-  constructor(private data: DataService, private dataTransferService : DataTransferService, private responsibilities: ResponsibilitiesComponent, private switchBoard: SwitchBoardService) { }
+  constructor(private data: DataService, private dataTransferService : DataTransferService,  public dialog: MatDialog) { }
+
+  openDialog(job: Job[]): void {
+    this.dialog.open(ResponsibilitiesComponent, {
+      data: { responsibility: job[0].responsibilities != " " ? job[0].responsibilities : "No responsibilities" }
+    });
+  }
+
+  openCapLeadDialog(): void {
```

Comment:
why dont you have a generic openDialog method which takes in any component and opens it instead of having a separate method that essentially does the same thing?